src/project.coffee

```diff
@@ -323,8 +323,22 @@ class Project extends Model
         # allow ENOENT errors to create an editor for paths that dont exist
         throw error unless error.code is 'ENOENT'
 
-    @bufferForPath(filePath).then (buffer) =>
-      @buildEditorForBuffer(buffer, options)
+    absoluteFilePath = @resolvePath(filePath)
+
+    fileSize = fs.getSizeSync(absoluteFilePath)
+
+    if fileSize >= 20 * 1048576 # 20MB
+      choice = atom.confirm
+        message: 'Atom can have issues with files over 20MB.'
```

Comment:
I like that the message is short :zap:, but wonder if we could tell the user a bit more about what's about to happen:

1. the performance might be reduced
2. some features will be disabled (e.g. syntax highlighting)

Here's a draft suggestion for discussion:

"This file if over 20MB and Atom performance might be reduced. In order to improve performance, some editing features will be disabled, e.g. syntax highlighting and soft wrap."